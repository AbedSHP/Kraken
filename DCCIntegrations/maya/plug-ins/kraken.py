import sys, json

from PySide import QtGui

import kraken.ui
from kraken.ui.kraken_ui import KrakenUI

from maya import cmds
# import maya.OpenMaya as OpenMaya
import maya.OpenMayaUI as OpenMayaUI
import maya.OpenMayaMPx as OpenMayaMPx

try:
  # Maya 2013 with custom pyside build
  import PySide.shiboken as shiboken
except:
  # Maya 2014 and higher
  import shiboken



# Command
class OpenKrakenEditorCommand(OpenMayaMPx.MPxCommand):
  def __init__(self):
    OpenMayaMPx.MPxCommand.__init__(self)

  # Invoked when the command is run.
  def doIt(self,argList):
    parent = OpenMayaUI.MQtUtil.mainWindow()
    parent = shiboken.wrapInstance(long(parent), QtGui.QWidget)
    dfgApp = KrakenUI(parent)

  # Creator
  @staticmethod
  def creator():
    return OpenMayaMPx.asMPxPtr( OpenKrakenEditorCommand() )


class KrakenUndoableCmd(OpenMayaMPx.MPxCommand):

  def __init__(self):
    OpenMayaMPx.MPxCommand.__init__(self)

  def isUndoable(self):
    return True

  def doIt(self, argList):
    return 0

  def redoIt(self):
    SingletonHost.getController().redo()
    return 0

  def undoIt(self):
    SingletonHost.getController().undo()
    return 0

  @staticmethod
  def creator():
    return OpenMayaMPx.asMPxPtr( KrakenUndoableCmd() )


# Initialize the script plug-in
def initializePlugin(mobject):
  mplugin = OpenMayaMPx.MFnPlugin(mobject)

  try:
    mplugin.registerCommand( 'openKrakenEditor', OpenKrakenEditorCommand.creator )
  except:
    sys.stderr.write( 'Failed to register DFG commands:openKrakenEditor' )
    raise

  try:
    mplugin.registerCommand( 'krakenUndoableCmd', KrakenUndoableCmd.creator )
  except:
    sys.stderr.write( 'Failed to register DFG commands:krakenUndoableCmd' )
    raise

# Uninitialize the script plug-in
def uninitializePlugin(mobject):
  mplugin = OpenMayaMPx.MFnPlugin(mobject)

  try:
    mplugin.deregisterCommand( 'openKrakenEditor' )
  except:
    sys.stderr.write( 'Failed to unregister command: openKrakenEditor' )


  try:
    mplugin.deregisterCommand( 'krakenUndoableCmd' )
  except:
    sys.stderr.write( 'Failed to unregister command: krakenUndoableCmd' )
