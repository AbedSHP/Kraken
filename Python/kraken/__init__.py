"""Kraken Framework."""

__all__ = ['core', 'examples', 'helpers', 'plugins', 'gui']