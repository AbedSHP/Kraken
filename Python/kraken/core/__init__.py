"""Kraken Core."""

VERSION_MAJOR = 0
VERSION_MINOR = 0
VERSION_BUILD = 1
VERSION_SUFFIX = "alpha"


def getVersion():
    """Contatenates the version globals and returns the current version of
    Kraken.

    Return:
    String, current version of Kraken.

    """

    versionString = str(VERSION_MAJOR) + "." + str(VERSION_MINOR) + "." + str(VERSION_BUILD)
    if VERSION_SUFFIX:
        versionString = versionString + "-" + VERSION_SUFFIX

    return versionString
