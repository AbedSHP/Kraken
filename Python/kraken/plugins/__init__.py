import os
import glob

__all__ = [os.path.splitext(os.path.basename(plugin))[0]
           for path in __path__
           for plugin in glob.glob(os.path.join(path, '*_plugin'))]


def getBuilder():
    """Returns the appropriate builder module for the DCC.

    Return:
    Builder, instance of the builder for the DCC.

    """

    dccBuilder = None

    for eachPlugin in __all__:
        mod = __import__("kraken.plugins." + eachPlugin, fromlist=['dccTest'])
        reload(mod)

        if mod.dccTest() is True:
            loaded_mod = __import__("kraken.plugins." + eachPlugin + ".builder", fromlist=['builder'])
            reload(loaded_mod)
            loaded_class = getattr(loaded_mod, 'Builder')

            dccBuilder = loaded_class()

    if dccBuilder is None:
        print "Failed to find DCC builder. Falling back to Python builder."

        from kraken.core import builder
        dccBuilder = builder.Builder()

    return dccBuilder


def getInspector():
    """Gets the inspector that belongs to the DCC calling this method.

    Return:
    Inspect, instance of the inspector for the DCC.

    """

    dccInspector = None

    for eachPlugin in __all__:
        mod = __import__("kraken.plugins." + eachPlugin, fromlist=['dccTest'])
        reload(mod)

        if mod.dccTest() is True:
            loaded_mod = __import__("kraken.plugins." + eachPlugin + ".inspector", fromlist=['inspector'])
            reload(loaded_mod)
            loaded_class = getattr(loaded_mod, 'Inspector')

            dccInspector = loaded_class()

    if dccInspector is None:
        print "Failed to find DCC inspector. Falling back to Python inspector."

        from kraken.core import inspector
        dccInspector = inspector.Inspector()

    return dccInspector