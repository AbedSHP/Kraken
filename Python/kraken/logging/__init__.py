
import logging

from kraken.plugins import getLogHandler


def getLogger(name):
    """Get's a logger and attaches the correct DCC compatible Handler.

    Args:
        name (str): Name of the logger to get / create.

    Returns:
        Logger: Logger.

    """

    dccHandler = getLogHandler()
    logger = logging.getLogger(name)
    logger.addHandler(dccHandler)

    return logger
