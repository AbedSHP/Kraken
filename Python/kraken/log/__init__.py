
import logging

from kraken.plugins import getLogHandler


def getLogger(name):
    """Get's a logger and attaches the correct DCC compatible Handler.

    Args:
        name (str): Name of the logger to get / create.

    Returns:
        Logger: Logger.

    """

    dccHandler = getLogHandler()
    logger = logging.getLogger(name)
    if dccHandler not in logger.handlers:
        logger.addHandler(dccHandler)

    return logger
