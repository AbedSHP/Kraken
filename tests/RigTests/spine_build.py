
from kraken import plugins
import kraken.examples.spine_component
reload(kraken.examples.spine_component)
from kraken.examples.spine_component import SpineComponent

spine = SpineComponent("spine")
builder = plugins.getBuilder()
builder.build(spine)
