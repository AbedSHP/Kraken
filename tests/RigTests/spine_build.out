Failed to find DCC builder. Falling back to Python builder.
Spine
Spine.controls
Spine.controls.Spine
Spine.controls.Spine.inputs
Spine.controls.Spine.outputs
Spine.controls.Spine.outputs.spine01
Spine.controls.Spine.outputs.spine02
Spine.controls.Spine.outputs.spine03
Spine.controls.Spine.outputs.spine04
Spine.controls.Spine.outputs.spine05
Spine.controls.Spine.outputs.spine06
Spine.controls.Spine.outputs.spineBase
Spine.controls.Spine.outputs.spineEnd
Spine.controls.Spine.cog
Spine.controls.Spine.cog.cog
Spine.controls.Spine.cog.cog.spine01
Spine.controls.Spine.cog.cog.spine01.spine01
Spine.controls.Spine.cog.cog.spine01.spine01.spine02
Spine.controls.Spine.cog.cog.spine01.spine01.spine02.spine02
Spine.controls.Spine.cog.cog.spine01.spine01.spine02.spine02.spine03
Spine.controls.Spine.cog.cog.spine01.spine01.spine02.spine02.spine03.spine03
Spine.controls.Spine.cog.cog.spine04
Spine.controls.Spine.cog.cog.spine04.spine04
Spine.deformers
Spine.deformers.Spine
Spine.deformers.Spine.spine01
Spine.deformers.Spine.spine02
Spine.deformers.Spine.spine03
Spine.deformers.Spine.spine04
Spine.deformers.Spine.spine05
Spine.deformers.Spine.spine06
